uber/templates/base.html

```diff
@@ -130,12 +130,64 @@
         var parseQueryString = function(queryString) {
             var params = {}, temp, i, len, queries = queryString.split('&');
             for (i = 0, len = queries.length; i < len; i++) {
+              if(queries[i]) {
                 temp = queries[i].split('=');
                 params[temp[0]] = temp[1];
+              }
             }
             return params;
         };
 
+        var replaceQueryParam = function(queryParam, value) {
+            var queryString = window.location.search.substring(1);
+            var params = parseQueryString(queryString);
+            if (_.has(params, queryParam) || value) {
+                if (value) {
+                    params[queryParam] = value;
+                } else {
+                    delete params[queryParam];
+                }
+                queryString = $.param(params);
+            }
+            if (queryString) {
+                queryString = '?' + queryString;
+            }
+            return queryString;
+        };
+
+        var replaceQueryParamInUrl = function(queryParam, value) {
+          try {
+              // Replace the query parameter in the web browser's address bar
+              if (history && history.replaceState) {
+                  var queryString = replaceQueryParam(queryParam, value);
+                  history.replaceState({}, document.title, window.location.pathname + queryString);
+                  return true;
+              }
+          } catch(ex) {
+              // Don't let any errors here disrupt the rest of the app!
+              try { console.log(ex); } catch(ex) {}
+          }
+          return false;
+        };
+
+        var removeHashFromUrl = function() {
```

Comment:
This... doesn't appear to be used anywhere, even in the panels version of this PR. 😅

Reply:
Yeah, this was used in an earlier version of the attractions feature where I was using the URL hash to preserve application state, like `uber/attractions#badge_num=123`. I eventually abandoned that idea, but I left this function in place because 1) it's a useful function and 2) I am lazy 😄